src/src/PluginStructs/P035_data_struct.h

```diff
@@ -0,0 +1,71 @@
+#ifndef PLUGINSTRUCTS_P035_DATA_STRUCT_H
+#define PLUGINSTRUCTS_P035_DATA_STRUCT_H
+
+#include "../../_Plugin_Helper.h"
+#ifdef USES_P035
+
+# include "../../ESPEasy-Globals.h"
+
+# include <ArduinoJson.h>
+# include <IRremoteESP8266.h>
+# include <IRac.h>
+# include <IRutils.h>
+# include <IRsend.h>
+
+extern void enableIR_RX(boolean enable); // To be found in _P016_IR.ino
+
+# define STATE_SIZE_MAX 53U
+# define PRONTO_MIN_LENGTH 6U
+
+# define from_32hex(c) ((((c) | ('A' ^ 'a')) - '0') % 39)
+
+# define P35_Ntimings 250u // Defines the ammount of timings that can be stored. Used in RAW and RAW2 encodings
+
+// # define P035_DEBUG_LOG // Enable for some (extra) logging
+
+struct P035_data_struct : public PluginTaskData_base {
+public:
+
+  P035_data_struct(int8_t gpioPin);
+
+  P035_data_struct() = delete;
+  ~P035_data_struct();
+
+  bool plugin_init(struct EventStruct *event);
+  bool plugin_exit(struct EventStruct *event);
+  bool plugin_write(struct EventStruct *event,
+                    const String      & string);
+
+private:
+
+  IRsend *Plugin_035_irSender = nullptr;
+
+  # ifdef P016_P035_Extended_AC
+
+  IRac          *Plugin_035_commonAc = nullptr;
+  stdAc::state_t st, prev;
```

Comment:
Shouldn't these be initialized?

Reply:
That type is a, rather elaborate, `struct` without a constructor, and the members are also not initialized. Not sure how to initialize them properly.

Reply:
Note of the IRac::initState(), which is technically a ctor for this struct
https://github.com/crankyoldgit/IRremoteESP8266/blob/93af543dcb849918b0c342ed1b6af68953a61d06/src/IRac.h#L71
Or just `st{};` to zero-init.

I suppose these structs *should've* had a proper ctor, but I completely forgot about this issue (at the end)
https://github.com/crankyoldgit/IRremoteESP8266/issues/1699

Reply:
So we can just incorporate the latest update of the `IRremoteESP8266` library to resolve this? (or apply this quick fix ourselves)

Reply:
If it is already fixed in the new version, you can also upgrade to the new version.
But maybe best to do that in a different PR, so if anything breaks we know where to look :)

@mcspr
Thanks for the link, full of good lessons/refreshers of C++.